lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -774,6 +774,26 @@ def _download_name(self, name, timeout=None, download_directory=None,
         sd_hash, file_path = yield helper.wait_or_get_stream(stream_info, lbry_file)
         defer.returnValue((sd_hash, file_path))
 
+    @defer.inlineCallbacks
+    def _publish_stream(self, name, bid, metadata, file_path=None, fee=None):
+        publisher = Publisher(self.session, self.lbry_file_manager, self.session.wallet)
+        verify_name_characters(name)
+        if bid <= 0.0:
+            raise Exception("Invalid bid")
+        if fee:
+            metadata = yield publisher.add_fee_to_metadata(metadata, fee)
+        if not file_path:
+            claim_out = yield publisher.update_stream(name, bid, metadata)
+        else:
+            claim_out = yield publisher.publish_stream(name, file_path, bid, metadata)
+            yield threads.deferToThread(reupload.reflect_stream, publisher.lbry_file)
```

Comment:
I'm not sure this needs to be a separate thread - I think as long you don't yield, you can create a deferred and have twisted process it.  This can probably be:
```
# make this its own deferred so as to not block publishing more than it already is
d = reupload.reflect_stream(publisher.lbry_file)
d.addCallback(lambda _: log.info('Successfully reflected %s after publish', publisher.lbry_file))
d.addErrback(log.fail(), 'Failed to reflect %s after publish', publisher.lbry_file))
```